### PR TITLE
feat: CreateCLPool & LinkCFMMtoCL pool into one gov-prop

### PR DESCRIPTION
**Pull request by stackman27**
Closes: [#5888](https://github.com/osmosis-labs/osmosis/issues/5888)

## What is the purpose of the change
Currently we have 2 gov prop message to migrate pool from GAMM.

Create CL Pool
Link GammPool with CL pool
This take 2 gov period which is too long. Lets create a gov-prop that will handle both of these commands in one

## Testing and Verifying
localosmosis tested 
```
/osmosis # 
/osmosis # osmosisd q gov proposals 1 
pagination:
  next_key: null
  total: "0"
proposals:
- content:
    '@type': /osmosis.gamm.v1beta1.CreateConcentratedLiquidityPoolsAndLinktoCFMMProposal
    description: add
    pool_records_with_cfmm_link:
    - balancer_pool_id: "2"
      denom0: uosmo
      denom1: uion
      exponent_at_price_one: "-6"
      spread_factor: "0.001000000000000000"
      tick_spacing: "100"
    title: add
  deposit_end_time: "2023-07-28T21:02:14.184663875Z"
  final_tally_result:
    abstain: "0"
    "no": "0"
    no_with_veto: "0"
    "yes": "0"
  is_expedited: false
  proposal_id: "1"
  status: PROPOSAL_STATUS_VOTING_PERIOD
  submit_time: "2023-07-26T21:02:14.184663875Z"
  total_deposit:
  - amount: "1000000000"
    denom: uosmo
  voting_end_time: "2023-07-26T21:03:34.234001152Z"
  voting_start_time: "2023-07-26T21:02:34.234001152Z"
-------
/osmosis # osmosisd q gamm cl-pool-link-from-cfmm 2
concentrated_pool_id: "4"
```

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/gamm/keeper/migrate.go [hunk 1]
```diff
@@ -355,3 +357,57 @@ func (k Keeper) UpdateMigrationRecords(ctx sdk.Context, records []gammmigration.
 	}
 	return nil
 }
+
+// CreateClPoolAndLinkToCFMMPool creates a CL pool and links it to an existing CFMM pool.
+func (k Keeper) CreateClPoolAndLinkToCFMMPool(ctx sdk.Context, records []types.PoolRecordWithCFMMLink) error {
```
Comment (stackman27) [inline x/gamm/keeper/migrate.go:362]: this is a repetitive code from updrades in v16 `createCanonicalConcentratedLiquidityPoolAndMigrationLink`. Would love suggestion on how we can use the same function on upgrade folder as well as the keeper folder here

Comment (ValarDragon) [reply x/gamm/keeper/migrate.go:362]: The upgrade folder can just import functions in the keeper folder

**Review by czarcas7ic — APPROVED** [review]
First pass looks good, can you test the prop in LO and post here if it works as intended?

**Comment by stackman27** [comment]
just checked 
```
/osmosis # osmosisd q gov proposal 1
content:
  '@type': /osmosis.gamm.v1beta1.CreateConcentratedLiquidityPoolsAndLinktoCFMMProposal
  description: add
  pool_records_with_cfmm_link:
  - balancer_pool_id: "2"
    denom0: uion
    denom1: uosmo
    exponent_at_price_one: "-6"
    spread_factor: "0.001000000000000000"
    tick_spacing: "100"
  title: add
deposit_end_time: "2023-08-04T20:35:26.728976737Z"
final_tally_result:
  abstain: "0"
  "no": "0"
  no_with_veto: "0"
  "yes": "500000000"
is_expedited: false
proposal_id: "1"
status: PROPOSAL_STATUS_PASSED
submit_time: "2023-08-02T20:35:26.728976737Z"
total_deposit:
- amount: "1000000000"
  denom: uosmo
voting_end_time: "2023-08-02T20:36:36.753489095Z"
voting_start_time: "2023-08-02T20:35:36.753489095Z"


/osmosis # osmosisd q gamm cl-pool-link-from-cfmm 2
concentrated_pool_id: "4"


/osmosis # osmosisd q concentratedliquidity pools
pagination:
  next_key: null
  total: "0"
pools:
- '@type': /osmosis.concentratedliquidity.v1beta1.Pool
  address: osmo1ad4r3uh5pdn5pgg5hnl6u5utfeqmpwstlvgvg2h2jdztrcnwkqgs3hs85z
  current_sqrt_price: "0.000000000000000000000000000000000000"
  current_tick: "0"
  current_tick_liquidity: "0.000000000000000000"
  exponent_at_price_one: "-6"
  id: "4"
  incentives_address: osmo1ra25mx8lch089mxf97nr6cnzns7vesecvw0mxj32fcaj5ygrsd8sqwhjpk
  last_liquidity_update: "2023-08-02T20:36:36.898121604Z"
  spread_factor: "0.001000000000000000"
  spread_rewards_address: osmo1w60ad6wuhkxeyuqfhnflvtkzcsrpey2xv6s05r00xxegl4vcedsqqu89g6
  tick_spacing: "100"
  token0: uion
  token1: uosmo



```

**Comment by stackman27** [comment]
> First pass looks good, it appears you only tested that the prop gets posted though according to your log though. Can you pass the prop and confirm the results? After that I think we can merge.

merging this as the gov prop has been tested and logs are posted

**Pull request merged**